lib/Models/WebMapServiceCatalogItem.js

```diff
@@ -319,18 +319,24 @@ WebMapServiceCatalogItem.prototype.updateFromCapabilities = function(capabilitie
 
     if (!defined(thisLayer)) {
         thisLayer = findLayers(capabilities.Capability.Layer, this.layers);
-        if (!defined(thisLayer)) {
-            return;
-        }
 
         if (defined(this.layers)) {
             var layers = this.layers.split(',');
             for (var i = 0; i < thisLayer.length; ++i) {
                 if (!defined(thisLayer[i])) {
-                    console.log('A layer with the name or ID \"' + layers[i] + '\" does not exist on the WMS Server - ignoring it.');
-                    thisLayer.splice(i, 1);
-                    layers.splice(i, 1);
-                    --i;
+                    if (thisLayer.length > 1) {
+                        console.log('A layer with the name or ID \"' + layers[i] + '\" does not exist on the WMS Server - ignoring it.');
+                        thisLayer.splice(i, 1);
+                        layers.splice(i, 1);
+                        --i;
+                    } else {
+                        throw new TerriaError({
+                            title: 'No layer found',
+                            message: 'The WMS dataset "' + this.name + '" has no layers matching "' + this.layers + '". (Perhaps it should be "' + capabilities.Capability.Layer.Layer.Name + '").' + 
+                            '\n\nEither the catalog file has been set up incorrectly, or the WMS server has changed.' + 
+                            '\n\nPlease report this error by sending an email to <a href="mailto:' + this.terria.supportEmail + '">' + this.terria.supportEmail + '</a>.'
```

Comment:
![screenshot 2016-02-24 12 02 28](https://cloud.githubusercontent.com/assets/678344/13272215/9b50b422-daee-11e5-9a59-29c48a2c61de.png)

You'd think so, but apparently not. We must do a replacement somewhere.